dashboard/app/models/teacher_based_experiment.rb

```diff
@@ -29,14 +29,25 @@ class TeacherBasedExperiment < Experiment
   validates :percentage, inclusion: 0..100
 
   def enabled?(user: nil, section: nil)
-    return false unless section
+    is_teacher = user.try(:teacher?)
+    return false unless section || is_teacher
+
+    if is_teacher
+      user_id = user.id
+      sections = user.sections
+    else
+      user_id = section.user_id
+      sections = [section]
+    end
+    user_id %= 100
```

Comment:
nit: Rather than mutate `user_id` (which really isn't changing), I think it is more readable to make a new `user_id_modulus` variable.